### DISCUSSION TITLE
Automated test suites for components

### DISCUSSION THREAD

**Discussion by MxGunderson** [RocketCommunicationsInc/astro#1200] [Q&A] [answered]
I'm new to Astro UXDS and i'm trying to figure out if Astro supports automated test suites? I can't seem to find any info on this topic outside of Astro Docs, which isn't the same as Astro UXDS, will automated test suites be supported at any point, if not already?

**Comment by micahjones13** [comment, marked as answer]
Hey @MxGunderson,

Most testing libraries should work with our components. I'd recommend using any that allow for interaction with shadow dom since our components use that. We use [playwright](https://playwright.dev/) for our component tests internally. and also have some [documentation around using React Testing Library](https://astro-components.netlify.app/?path=/story/frameworks-react--page#testing) in case that's what you're trying to use. Is there a specific testing library you want to use, or anything you'd like to see us ship out of the box for testing?
